### PR TITLE
Loop protection in HTTPS upgrades

### PR DESCRIPTION
**Pull request by kdzwinel**
Domain is HTTPS upgradable but page will constantly try to downgrade to HTTP. Client should detect that and allow page to load. Unfortunately our extension seems to have a bug and doesn't do it.

https://app.asana.com/0/1186013049913869/1201057222436407/f

### CODE REVIEW

File: privacy-protections/https-loop-protection/index.html [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>HTTPS upgrade loop protection</title>
+
+    <script src='./main.js' defer></script>
+    <link href='./style.css' rel='stylesheet'></link>
+</head>
+<body>
+    <p><a href="../../">[Home]</a> ↣ <a href="../">[Privacy Protections Tests]</a> ↣ <strong>[HTTPS Upgrade Loop Protection]</strong></p>
+
+    <p>This tst will navigate to page on a https upgradable domain that imadiatelly redirects to its http version. This will cause a http↔http loop (client trying to upgrade, page downgrading). Clients should detect this scenario and allow page to load.</p>
```
Comment (jonathanKingston) [inline privacy-protections/https-loop-protection/index.html:14]:
```suggestion
    <p>This test will navigate to page on a https upgradable domain that immediately redirects to its http version. This will cause a http↔http loop (client trying to upgrade, page downgrading). Clients should detect this scenario and allow page to load.</p>
```

File: privacy-protections/https-upgrades/main.js [hunk 2]
```diff
@@ -174,7 +174,7 @@ function downloadTheResults () {
     const a = document.createElement('a');
     const url = window.URL.createObjectURL(new Blob([data], { type: 'application/json' }));
     a.href = url;
-    a.download = 'fingerprinting-results.json';
+    a.download = 'https-upgrades-results.json';
```
Comment (kdzwinel) [inline privacy-protections/https-upgrades/main.js:177]: unrelated change - renaming downloaded file to match test name

File: privacy-protections/https-loop-protection/http-only.html [hunk 3]
```diff
@@ -0,0 +1,36 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>HTTPS upgrade loop protection</title>
+</head>
+<body>
+    <script>
+        const MAX = 15;
```
Comment (kdzwinel) [inline privacy-protections/https-loop-protection/http-only.html:10]: if loop protection doesn't kick in after 15 attempts - give up

File: privacy-protections/https-loop-protection/main.js [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+const testsDiv = document.querySelector('#tests');
+const testsSummaryDiv = document.querySelector('#tests-summary');
+const testsDetailsDiv = document.querySelector('#tests-details');
+
+const TEST_DOMAIN = 'good.third-party.site';
+
+const tests = [
+    {
+        id: 'upgrade-navigation',
+        run: () => {
```
Comment (kdzwinel) [inline privacy-protections/https-loop-protection/main.js:13]: most of this file, besides the contents of this test, is just a template. We should probably extract it to some reusable helpers instead of copying it over and over

**Review by jonathanKingston — APPROVED** [review]
This is really great, thanks for changing the wording to make the test clearer :). Also thanks for the PR comments yourself, I think this helps often.
